web/js/codeworld_shared.js

```diff
@@ -70,6 +70,72 @@ var Alert = (() => {
   return mine;
 })();
 
+var hintBlacklist = [
+    // Symbols that only exist to implement RebindableSyntax or map to
+    // built-in Haskell types.
+    "Bool",
+    "IO",
+    "fail",
+    "fromCWText",
+    "fromDouble",
+    "fromInt",
+    "fromInteger",
+    "fromRational",
+    "fromString",
+    "ifThenElse",
+    "toCWText",
+    "toDouble",
+    "toInt",
+
+    // Deprecated exports.
+    "MousePress",
+    "MouseRelease",
+    "MouseMovement",
+    "MouseButton",
+    "LeftButton",
+    "RightButton",
+    "MiddleButton",
+    "Maybe",
+    "Nothing",
+    "Just",
+    "withDefault",
+    "hasValue",
+    "definitely",
+    "path",
+    "thickPath",
+    "text",
+    "styledText",
+
+    // Experimental exports.
+    "White",
+    "Black",
+    "Gray",
+    "Grey",
+    "Red",
+    "Orange",
+    "Yellow",
+    "Green",
+    "Blue",
+    "Purple",
+    "Pink",
+    "Brown",
+    "Pi",
+];
+
+var codeWorldBuiltinsDocs = {};
+
+function onHover(cm, data, node){
+    var result = null;
+    if (data && data.token && data.token.string != "") {
+        var token_name = data.token.string;
+        if (token_name && hintBlacklist.indexOf(token_name) == -1 && codeWorldBuiltinsDocs[token_name]){
```

Comment:
[nit] If you're going to test data.token.string for truthiness, why not do it above where you currently compare with ""?

Reply:
Btw, what is "[nit]"?  Deleted `!= ""`

Reply:
Oh, [nit] means I'm admittedly nitpicking.  Basically, I mean if you feel the original is better, it's not something I'd push back on.